dali/operators/nvcvop/nvcvop.h

```diff
@@ -0,0 +1,225 @@
+ *
+ * @param border_mode border mode name
+ */
+NVCVBorderType GetBorderMode(std::string_view border_mode);
+
+/**
+ * @brief Get nvcv data kind of a given data type
+ */
+nvcv::DataKind GetDataKind(DALIDataType dtype);
+
+/**
+ * @brief Get image format for an image with a given data type and number of channels.
+ *
+ * The memory layout of the image is assumed to be contiguous, with interleaved, equally-sized channels.
+ */
+nvcv::ImageFormat GetImageFormat(DALIDataType dtype, int num_channels);
+
+/**
+ * @brief Wrap a sample view into an nvcv Image
+ *
+ * @param sample sample view
+ * @param format image format. It needs to match the shape and data type of the sample view
+ */
+nvcv::Image AsImage(SampleView<GPUBackend> sample, const nvcv::ImageFormat &format);
+
+/**
+ * @brief Wrap a const sample view into an nvcv Image
+ *
+ * @param sample sample view
+ * @param format image format. It needs to match the shape and data type of the sample view
+ */
+nvcv::Image AsImage(ConstSampleView<GPUBackend> sample, const nvcv::ImageFormat &format);
+
+/**
+ * @brief Allocates an image batch using a dynamic scratchpad.
+ * Allocated images have the shape and data type matching the samples in the given TensorList.
+ *
+ * It can be used to allocate a workspace batch that matches a given input batch.
+ *
+ * The output images are valid only for the lifetime of the scratchpad.
+ */
+void AllocateImagesLike(nvcv::ImageBatchVarShape &output, const TensorList<GPUBackend> &t_list,
+                        kernels::Scratchpad &scratchpad);
+
+/**
+ * @brief Wrap samples from the given TensorList into nvcv Images and push them to an output Image batch
+ * The resulting images cannot outlive the TensorList.
+ */
+void PushImagesToBatch(nvcv::ImageBatchVarShape &batch, const TensorList<GPUBackend> &t_list);
+
+/**
+ * @brief A base class for the CVCUDA operators.
+ * It adds convenience methods to access inputs, outputs and arguments as nvcv types.
+ */
+template <typename BaseOp>
+class NVCVOperator: public BaseOp {
+ public:
+  using BaseOp::spec_;
+  using BaseOp::BaseOp;
+
+  /**
+   * @brief Get an operator argument as an nvcv Tensor.
+   * This method uses the dynamic scratchpad to allocate the (pinned and device) buffers
+   * used by the output tensor.
+   *
+   * The scratchpad should return its memory in an order that is compatible with ws.stream().
+   * The output is valid only for the lifetime of the scratchpad.
+   *
+   * @tparam T data type of the DALI operator argument
+   * @tparam DTYPE expected nvcv Tensor data type
+   * @param arg_shape shape of the DALI operator argument.
+   */
+  template <typename T, NVCVDataType DTYPE>
+  nvcv::Tensor AcquireTensorArgument(Workspace &ws, kernels::Scratchpad &scratchpad,
+                                     ArgValue<T, 1> &arg, const TensorShape<> &arg_shape,
+                                     TensorLayout layout = "") {
+    auto dtype = nvcv::DataType(DTYPE);
```

Comment:
Why compile-time DTYPE?
```suggestion
  template <typename T>
  nvcv::Tensor AcquireTensorArgument(Workspace &ws, kernels::Scratchpad &scratchpad,
                                     ArgValue<T, 1> &arg, const TensorShape<> &arg_shape,
                                     nvcv::DataType dtype,
                                     TensorLayout layout = "") {
```
Also, we could probably have some default inferred from T.

Reply:
done